src/App.tsx

```diff
@@ -1,14 +1,68 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { SelectOptions } from './types/SelectOptions';
+import { getTodos, getUser } from './api';
 
 export const App: React.FC = () => {
+  // #region initial todos and loading
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getTodos()
+      .then(tds => {
+        const promises = tds.map(todo => {
+          return getUser(todo.userId).then(user => ({
+            ...todo,
+            user,
+          }));
+        });
+
+        return Promise.all(promises);
+      })
+      .then(todosWithUsers => {
+        setTodos(todosWithUsers);
+        setIsLoading(false);
+      })
+      .catch(error => {
+        throw new Error('Error fetching todos or users:', error);
+      });
```

Comment:
The code to fetch todos and users is directly interacting with the DOM. It's recommended to use React as much as possible instead.

Reply:
but it''s not using DOM